sdk/web-pubsub/web-pubsub-client/src/webPubSubClientCredential.ts

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { GetClientAccessUrlOptions } from "./models";
+
+/**
+ * Type to be used to get clientAccessUri
+ */
+export type ClientAccessUriProvider = (options?: GetClientAccessUrlOptions) => Promise<string>;
+
+/**
+ * The WebPubSubClient credential
+ */
+export interface WebPubSubClientCredential {
+  /**
+   * Gets an `getClientAccessUrl` which is used in connecting to the service
+   * @param abortSignal - An implementation of `AbortSignalLike` to cancel the operation.
+   */
+  getClientAccessUrl(options?: GetClientAccessUrlOptions): Promise<string>;
+}
+
+/**
+ * Default implementation of WebPubSubClientCredential
+ */
+export class DefaultWebPubSubClientCredential implements WebPubSubClientCredential {
```

Comment:
We use "credential" for authentication. What is the purpose of this class?

Reply:
Just like the comment above. It returns a url with jwt to authenticate with the service